### PR TITLE
[asn] Fix #27 - extend string encoding options of fromJSON API.

### PR DESCRIPTION
**Pull request by turon**
Fix #27 
Adds support for flexible string encoding options when creating a certificate with the `fromJSON` API.

### CODE REVIEW

File: src/name.ts [hunk 1]
```diff
@@ -265,6 +265,18 @@ export class Name {
           } else {
             if (typeId === this.getName("E") || typeId === this.getName("DC")) {
               asnAttr.value.ia5String = value;
+            } else if (typeof value === "object") {
+                const entry = value as object;
```
Comment (turon) [inline src/name.ts:269]: Typescript compiler seemed to want this cast conversion, even though the `typeof` check above ensures it.

Comment (WorldThirteen) [reply src/name.ts:269]: I guess the reason is that `typeof value === "object"` will be true for value = null.
So you may add a null check to the if statement, or process this case separately, to ensure the null value case is handled correctly.

Comment (turon) [reply src/name.ts]: Adjusted JsonAttributeAndValue and order of operations for stricter type control.

**Comment by microshine** [comment]
@turon The new version `@peculiar/x509@1.8.2` has been published

Thank you for your update

**Pull request merged**